api/app/utils/playground.ts

```diff
@@ -0,0 +1,134 @@
+import { Parser, AST, Select } from "node-sql-parser";
+
+import { BadParamsError, SQLExecuteError } from "../core/Query";
+
+export class SqlParser {
+  parser: Parser;
+  sql: string;
+  ast: AST;
+  type: string;
+  id: number;
+
+  constructor(type: "repo" | "user", id: string, sql: string) {
+    this.parser = new Parser();
+    this.sql = sql;
+    switch (type) {
+      case "repo":
+        this.type = "repo_id";
+        break;
+      case "user":
+        this.type = "actor_id";
+      default:
+        throw new BadParamsError("playground", `invalid type ${type}`);
+    }
+    this.id = parseInt(id, 10); // consider bigint
+    this.ast = this.sqlToAst(sql);
+  }
+
+  private sqlToAst(sqlString: string) {
+    try {
+      const ast = this.parser.astify(sqlString);
+      if (Array.isArray(ast) && ast.length > 1) {
+        throw new BadParamsError(
+          "playground",
+          "only one sql statement is allowed"
+        );
+      } else if (Array.isArray(ast)) {
+        return ast[0];
+      } else {
+        return ast;
+      }
+    } catch (error) {
+      throw new BadParamsError("playground", "invalid sql");
+    }
+  }
+
+  private validateAst(ast: AST) {
```

Comment:
what about sub query

Reply:
Allow all nested `SELECT` commands.
For current design, only allow `SELECT`(nested command supported) and `DESC`.